runtime/vm/extendedMessageNPE.cpp

```diff
@@ -374,30 +519,1493 @@ getCompleteNPEMessage(J9VMThread *vmThread, U_8 *bcCurrentPtr, J9ROMClass *romCl
+}
+
+/**
+ * Get a fully qualified method name.
+ *
+ * Note: the caller is responsible for freeing the returned string if it is not NULL.
+ *
+ * @param[in] vmThread current J9VMThread
+ * @param[in] romClass the romClass containing the method
+ * @param[in] bcIndex the bytecode byte stream where the method is to be computed
+ *
+ * @return a char pointer to a fully qualified method name,
+ *         NULL if not successful, but keep application exception instead of throwing OOM.
+ */
+static char*
+getFullyQualifiedMethodName(J9VMThread *vmThread, J9ROMClass *romClass, U_8 *bcIndex)
+{
+	UDATA bc = *bcIndex;
+	UDATA index = PARAM_16(bcIndex, 1);
+	J9ROMConstantPoolItem *constantPool = J9_ROM_CP_FROM_ROM_CLASS(romClass);
+
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	Trc_VM_GetFullyQualifiedMethodName_Entry(vmThread, bc, index, bcIndex);
+	if (JBinvokestaticsplit == bc) {
+		index = *(U_16 *)(J9ROMCLASS_STATICSPLITMETHODREFINDEXES(romClass) + index);
+	} else if (JBinvokespecialsplit == bc) {
+		index = *(U_16 *)(J9ROMCLASS_SPECIALSPLITMETHODREFINDEXES(romClass) + index);
+	}
+
+	J9ROMMethodRef *info = (J9ROMMethodRef *) &constantPool[index];
+	J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(info);
+	J9UTF8 *methodName = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+	J9UTF8 *methodSig = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+	J9UTF8 *className = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *) &constantPool[info->classRefCPIndex]);
+	char *fullyQualifiedClassName = convertToJavaFullyQualifiedName(vmThread, className);
+	char *methodSigParameters = convertMethodSignature(vmThread, methodSig);
+	char *result = NULL;
+
+	if ((NULL != fullyQualifiedClassName) && (NULL != methodSigParameters)) {
+		result = getMsgWithAllocation(vmThread, "%s.%.*s%s",
+			fullyQualifiedClassName, J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), methodSigParameters);
+	}
+	j9mem_free_memory(fullyQualifiedClassName);
+	j9mem_free_memory(methodSigParameters);
+
+	Trc_VM_GetFullyQualifiedMethodName_Exit(vmThread, index, info, fullyQualifiedClassName, methodSigParameters, result);
+	return result;
+}
+
+/**
+ * Get a local name, the actual variable name,
+ * parameter# if the temp wasn't changed, or local# if the temp has been changed.
+ *
+ * Note: the caller is responsible for freeing the returned string if it is not NULL.
+ *
+ * @param[in] vmThread current J9VMThread
+ * @param[in] romMethod the romMethod containing the local
+ * @param[in] localVar the local variable index
+ * @param[in] bcIndex the bytecode index
+ * @param[in] temps the location of bytecode temps supplied via liveStack->stackElements
+ *
+ * @return a char pointer to a local name, NULL if not successful
+ */
+static char*
+getLocalsName(J9VMThread *vmThread, J9ROMMethod *romMethod, U_16 localVar, UDATA bcIndex, UDATA *temps)
+{
+	J9MethodDebugInfo *methodInfo = getMethodDebugInfoFromROMMethod(romMethod);
+	char *result = NULL;
+	BOOLEAN foundName = FALSE;
```

Comment:
As this is a `.cpp` file, please use `bool` instead of `BOOLEAN` throughout.

Reply:
Fixed.